nixneovim.nix

```diff
@@ -151,6 +157,8 @@ in
         plugins = cfg.extraPlugins;
       };
 
+      environment.variables.EDITOR = mkIf cfg.defaultEditor (mkOverride 900 "nvim");
```

Comment:
Why is this needed? Isn't this handled by the `programs.neovim.defaultEditor` config?

Reply:
Its the same method used to set the `programs.neovim.defaultEditor`  but it eliminates redundancy since `nixneovim` already installs neovim.

Before:
```nix
    programs.neovim.defaultEditor = true;
    programs.nixneovim = {
       ...
     }
```
After:
```nix
    programs.nixneovim = {
        defaultEditor = true;
        ...
    }
```

Reply:
You set the `programs.neovim.defaultEditor` to `cfg.defaultEditor`.
However, `programs.neovim.defaultEditor` already sets the `EDITOR` variable` as seen [here](https://github.com/NixOS/nixpkgs/blob/5dab6490fe6d72b3f120ae8660181e20f396fbdf/nixos/modules/programs/neovim.nix#L166).

Therefore, It should not be necessary to set the `EDITOR` variable again, here. Right?

Reply:
Alright I understand what your saying, no need to recreate the variable since `programs.neovim.defaultEditor` already creates it.